recsys/utils.py

```diff
@@ -1,47 +1,92 @@
 import calendar
+from datetime import datetime
 import gdown
 import gzip
 import json
+import logging
 import os
+from typing import Any, Dict, List, Optional, Tuple
+
+import numpy as np
 import pandas as pd
-from datetime import datetime
+import pickle
 from sklearn.model_selection import train_test_split
 from tensorflow.keras import backend as K
 
+logging.basicConfig(level=logging.INFO)
 
-def maybe_download_files():
-    if not os.path.exists("data"):
-        os.makedirs("data", exist_ok=True)
-        os.chdir("data")
-        # Books
-        gdown.download(
-            "https://drive.google.com/uc?id=1gH7dG4yQzZykTpbHYsrw2nFknjUm0Mol",
-            output=None,
-            quiet=None,
-        )
-        # Interactions
-        gdown.download(
-            "https://drive.google.com/uc?id=1NNX7SWcKahezLFNyiW88QFPAqOAYP5qg",
-            output=None,
-            quiet=None,
-        )
-        # Reviews
-        gdown.download(
-            "https://drive.google.com/uc?id=1M5iqCZ8a7rZRtsmY5KQ5rYnP9S0bQJVo",
-            output=None,
-            quiet=None,
-        )
-        os.chdir("..")
+IS_TRAVIS = os.environ.get("TRAVIS") == "true"
+
+YA_BOOKS_URL = "https://drive.google.com/uc?id=1gH7dG4yQzZykTpbHYsrw2nFknjUm0Mol"
+YA_INTERACTIONS_URL = "https://drive.google.com/uc?id=1NNX7SWcKahezLFNyiW88QFPAqOAYP5qg"
+YA_REVIEWS_URL = "https://drive.google.com/uc?id=1M5iqCZ8a7rZRtsmY5KQ5rYnP9S0bQJVo"
+SMALL_DATA_URL = "https://drive.google.com/uc?id=1_UY4xTbk3o0xjGbVllQZC2bBt-WAwyF_"
+
+BOOK_DATA = "data/goodreads_books_young_adult.json.gz"
+INTERACTIONS_DATA = "data/goodreads_interactions_young_adult.json.gz"
+REVIEWS_DATA = "data/goodreads_reviews_young_adult.json.gz"
+SAMPLE_DATA = "data/sample_data.pkl"
```

Comment:
hmmmm, can we go back to the smaller json.gz versions? more authentic test of the data utils

Reply:
The trouble with that is, It's hard to get books, reviews, and interactions that have high enough intersection with each other. If we just sample ~ 1000 of each, we end up with none of the dev interactions having reviews, most interactions having no book metadata, and so on. So LFs get 0 coverage, and loss becomes nan.